cwms-data-api/src/main/java/cwms/cda/data/dto/location/kind/Outlet.java

```diff
@@ -20,31 +20,67 @@
 
 package cwms.cda.data.dto.location.kind;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 import com.fasterxml.jackson.databind.PropertyNamingStrategies;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import cwms.cda.data.dto.CwmsDTOValidator;
 import cwms.cda.data.dto.CwmsId;
 import cwms.cda.data.dto.Location;
 import cwms.cda.formatters.Formats;
 import cwms.cda.formatters.annotations.FormattableWith;
 import cwms.cda.formatters.json.JsonV1;
+import java.util.ArrayList;
+import java.util.List;
 
 @FormattableWith(contentType = Formats.JSONV1, formatter = JsonV1.class, aliases = {Formats.DEFAULT, Formats.JSON})
 @JsonDeserialize(builder = Outlet.Builder.class)
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
-@JsonPropertyOrder({ "projectId", "location" })
+@JsonPropertyOrder({"projectId", "location"})
 public class Outlet extends ProjectStructure {
+    private final List<CompoundOutletRecord> compoundOutletRecords = new ArrayList<>();
+    private final String ratingGroupId;
 
     private Outlet(Builder builder) {
         super(builder.projectId, builder.location);
+        ratingGroupId = builder.ratingGroupId;
+        if (builder.compoundOutletRecords != null) {
+            compoundOutletRecords.addAll(builder.compoundOutletRecords);
+        }
+    }
+
+    public List<CompoundOutletRecord> getCompoundOutletRecords() {
+        return new ArrayList<>(compoundOutletRecords);
+    }
+
+    public String getRatingGroupId() {
+        return ratingGroupId;
+    }
+
+    @Override
+    protected void validateInternal(CwmsDTOValidator validator) {
+        super.validateInternal(validator);
+        compoundOutletRecords.forEach(CompoundOutletRecord::validate);
```

Comment:
validate will fail on first validation fail instead of iterating over each. I added `validator.validateCollection` to help with this.

Reply:
Updated